samples/MvcApp/App_Start/Startup.Auth.cs

```diff
@@ -18,6 +21,18 @@ public void ConfigureAuth(IAppBuilder app)
             app.CreatePerOwinContext<ApplicationUserManager>(ApplicationUserManager.Create);
             app.CreatePerOwinContext<ApplicationSignInManager>(ApplicationSignInManager.Create);
 
+            // These must match the data protection settings in MvcCoreApp Program.cs for cookie sharing to work
+            var sharedApplicationName = "CommonMvcAppName";
+            var sharedDataProtectionProvider = DataProtectionProvider.Create(
+                // This directory is used to share dataprotection keys between MvcApp and MvcCoreApp
+                new DirectoryInfo(Path.Combine(Path.GetTempPath(), "sharedkeys", sharedApplicationName)),
+                    builder => builder.SetApplicationName(sharedApplicationName))
+                    .CreateProtector(
+                        "Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationMiddleware",
+                        // Must match the Scheme name on the MvcCoreApp, i.e. IdentityConstants.ApplicationScheme
+                        "SharedCookie",
+                        "v2");
+
```

Comment:
@mjrousos I cleaned things up to remove a lot of the extraneous settings, while we could still add a ConfigureSharedCookie("appName", "sharedKeyPath", "authenticationScheme", cookieAuthenticationOptions), this at least makes the changes into 2 logical sections (configure the data protection, and then use it for the cookie ticket format.  Thoughts ?

Reply:
Perhaps its good enough with just a single helper to hide the hardcoded things out of their control with a `var sharedDataProtectionProvider = CreateSharedDataProtectionProvider(sharedKeyDirectory, appname, authScheme)`

Reply:
I'm liking this better - a `ConfigureSharedCookie` method seems easy to implement and would hide even more of the stuff people don't need to muck with (i.e. do they need access to the data protection provider?). Someone could always build it up themselves, but simplifying the set up as much as possible would be nice (espeically to get rid of any hard coded things)

Reply:
I like the idea of minimizing the code the customer has to write, but unfortunately I think it will be common to need access to the data protection provider. A shared directory won't work for a lot of scenarios (anything where the ASP.NET and ASP.NET Core apps run on different machines. App Service?) so I think it's important that customers be able to provide their own data protection providers (which may use Redis or blob storage or something as a backing store).

Can we have a helper method that will make the `CreateProtector` call with all the right string constants, but allow the user to pass in a `DataProtectionProvider`?

Maybe an API like `ConfigureSharedCookie(this IAppBuilder app, CookieAuthenticationOptions options, IDataProtectionProvider dataProtectionProvider)` or something like that. It would mean pulling in a reference to OWIN stuff in the adatpers library, but we could split this API out into a separate package that users would only depend on if they wanted this sort of shared auth enabled (in which case they were using OWIN anyhow).

Reply:
See #2  - it could be put in there.